WordPress/src/main/java/org/wordpress/android/ui/posts/QuickStartPromptDialogFragment.kt

```diff
@@ -92,7 +92,7 @@ class QuickStartPromptDialogFragment : AppCompatDialogFragment() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        this.siteRecord = SiteRecord(selectedSiteRepository.getSelectedSite())
+        this.siteRecord = SiteRecord(selectedSiteRepository.getSelectedSite()!!)
```

Comment:
I think the change here does not introduce a new bug but the need of a forced unwrap due to the conversion of the `SiteRecord` in Kotlin might indicate that we need a null check or a safe call here. I've noticed bugs in the past related to the `SiteModel` not being fetched correctly. Wdyt?

Reply:
Nice point! The null check will be like the following code snippet:

```
if (selectedSiteRepository.getSelectedSite() != null) {
    this.siteRecord = SiteRecord(selectedSiteRepository.getSelectedSite())
}
```
This change will cause `Caused by kotlin.UninitializedPropertyAccessException: lateinit property binding has not been initialized` later if the `this.siteRecord` is not set.

I think it is an edge case that was not handled during the development. I'd suggest to display an error message and dismiss the DialogFragment directly.

```
if (selectedSiteRepository.getSelectedSite() == null) {
    toastErrorMessage()
    dismiss()
    return
}
```

Reply:
> I think it is an edge case that was not handled during the development. I'd suggest to display an error message and dismiss the DialogFragment directly if that case happened because there's nothing we can display if selectedSiteRepository.getSelectedSite() == null.

I agree it is an edge case and the proposed solution sounds good 👍